pkg/metric_storage/vault/vault.go

```diff
@@ -58,7 +60,10 @@ func (v *GroupedVault) GetOrCreateGaugeCollector(name string, labelNames []strin
 			return nil, fmt.Errorf("gauge '%s' %v registration: %v", name, labelNames, err)
 		}
 		v.collectors[name] = collector
+	} else if !LabelNamesEqual(collector.LabelNames(), labelNames) {
+		collector.UpdateLabels(labelNames)
```

Comment:
not sure how to make it less noisy, maybe it makes sense to store all the collector's labels in a map alongside the slice so that it would be faster to check if any new labels are provided